enterprise/backend/src/metabase_enterprise/audit_app/permissions.clj

```diff
@@ -64,11 +64,15 @@
   [group-id changes]
   (let [[change-id type] (first (filter #(= (first %) (:id (default-audit-collection))) changes))]
       (when change-id
-        (let [permission-value (case type
-                                 :read  :unrestricted
-                                 :none  :no-self-service
-                                 :write (throw (ex-info (tru (str "Unable to make audit collections writable."))
-                                                        {:status-code 400})))
+        (let [data-access-value (case type
+                                  :read  :unrestricted
+                                  :none  :no-self-service
+                                  :write (throw (ex-info (tru (str "Unable to make audit collections writable."))
+                                                         {:status-code 400})))
+              create-queries-value (case type
+                                     :read :query-builder
+                                     :none :no)
               view-tables         (t2/select :model/Table :db_id perms/audit-db-id :name [:in audit-db-view-names])]
           (doseq [table view-tables]
-            (data-perms/set-table-permission! group-id (u/the-id table) :perms/data-access permission-value))))))
+            (data-perms/set-table-permission! group-id table :perms/data-access data-access-value)
```

Comment:
TODO: I just noticed I'm not setting `:perms/read-data` here, and I think I should be?

Reply:
Hmm, actually, I don't think so. The previous behavior would only ever create `:unrestricted` or `:no-self-service` `:data-access` which are both mapped to `:unrestricted` `:view-data`.

i.e. we're only controlling audit DB access via query perms. nobody should have `:blocked` data perms, particularly since we can't guarantee that an EE instance with the audit feature flag also has advanced permissions enabled.

Reply:
Oh right, thanks. I think I thought about this when I implemented it and then forgot about it when I was looking back over things.